proof/crefine/RISCV64/Finalise_C.thy

```diff
@@ -1454,10 +1459,17 @@ lemma suspend_ccorres:
     apply clarsimp
     apply (rule conseqPre, vcg)
     apply (rule subset_refl)
-   apply (rule_tac Q="\<lambda>_. invs' and tcb_at' thread" in hoare_post_imp)
-    apply fastforce
-   apply (wpsimp wp: hoare_vcg_all_lift)
-  by (auto simp: ThreadState_defs)[1]
+   apply (rule hoare_strengthen_post)
+    apply (rule hoare_vcg_conj_lift)
+     apply (rule hoare_vcg_conj_lift)
+      apply (rule hoare_vcg_conj_lift)
+       apply (rule cancelIPC_sch_act_simple)
+      apply (rule cancelIPC_tcb_at'_better[where P=id, simplified, where p=thread])
+     apply (rule cancelIPC_invs')
+    apply (rule cancelIPC_weak_sch_act_wf)
```

Comment:
I usually would want to avoid explicit instantiation as was here before, but the previous version does look more stable and automatic. Would adapting the instantiation here be a very big term or otherwise ugly?

Reply:
I think it might just be adding `weak_sch_act_wf (ksSchedulerAction s) s` to what was there before, but I can't remember exactly. I can investigate

Reply:
Yep, I just needed to add that, and also `sch_act_simple`, and then I could follow the old proof. Updated in the last commit, which I'll squash

Reply:
Cool, thanks, I do like the new version better.